### ISSUE TITLE
Loss is Nan

### ISSUE THREAD

**Issue by mahirahzaini** [qc17-THU/DL-SR#15]
Hi,

I'm trying to reproduce your training result for CCPs dataset. I tried setting up the batch size for DFCAN to 4 and batch size for DFGAN to 2 but during training at around 9000 epochs or so the loss is Nan. I'm just wondering what parameters do you use to train CCPs dataset and how can I avoid getting Nan for my loss? I tried to google this and most people said to check if the preprocessing dataset is corrupted. I then checked this but did not see any corrupted data in my dataset.

Thank you in advance.

**Comment by qc17-THU** [comment]
Hi,
 you can try to use a smaller initial learning rate, such as 1e-5.

**Comment by mahirahzaini** [comment]
Thank you for the update. I’m also wondering how do you calculate the mini batch iteration used for the training of ccps dataset in your supplementary materials. From my understanding minibatch iteration is number of samples divided by batch size. Is this the same calculation you’re using?

**Comment by qc17-THU** [comment]
Hi, 
The mini batch iteration was select empirically but not by calculation to ensure that the models were trained properly.

**Comment by mahirahzaini** [comment]
Okay and How do I implement this minibatch to the training?

**Comment by mahirahzaini** [comment]
How do you calculate the minibatch in the plot you showed in the supplementary materials for CCPs dataset? As far as I know the code you provided only give epochs in this case it’s iterations

**Comment by qc17-THU** [comment]
The 'epoch' in the codes actually should be 'minibatch iteration'. I have modified this error just now. 
This 'minibatch iteration' corresponds to the plot showed in the supplementary materials.
You can change the argument 'iteration' to apply this minibatch to the training.

**Comment by mahirahzaini** [comment]
Thank you for the clarification. I also run the preprocessing using my own dataset. After training the model with my dataset the output of the image appears to be stacked in the corner of the image. It looks like there are multiple borders stacked in the corner of the image. Do you know what this could possibly be? have you seen this before?

Thank you.

**Comment by qc17-THU** [comment]
I haven't seen this before. Can you show me some output images here?

**Comment by mahirahzaini** [comment]
for sure. I've attached the images below along with the ground truth image and low-resolution image.
![dfcan_dg_output](https://user-images.githubusercontent.com/84991393/179270046-ffe457b0-5b29-4561-9f81-eb96a89bb64f.png)

**Comment by qc17-THU** [comment]
I didn't notice artifacts like these in my experiments, but I think maybe it is caused by the usage of the pixel shuffle layer. Maybe you can try replacing pixel shuffle layer with upsampling layer to fix it.

**Comment by mahirahzaini** [comment]
In supplementary Figure 13 e you have Training loss , mini batch iteration and validation NRMSE plotted for DFCAN. For the training loss is it calculated using MSE + lambda x (1 - SSIM) ?

**Comment by qc17-THU** [comment]
Yes, the training loss shown in Supp. Fig. 13e is calculated using MSE + 0.1 * (1-SSIM).

**Comment by qc17-THU** [comment]
> for sure. I've attached the images below along with the ground truth image and low-resolution image. ![dfcan_dg_output](https://user-images.githubusercontent.com/84991393/179270046-ffe457b0-5b29-4561-9f81-eb96a89bb64f.png)

About this problem, I wonder if you use the original low resolution images as network inputs or use the upsampling images (e.g. by bicubic interpolation) as sampling.

**Comment by mahirahzaini** [comment]
I believe I use the original low-resolution images as network input.

**Comment by mahirahzaini** [comment]
Thank you so much for taking the time to respond to me.

**Comment by qc17-THU** [comment]
> I believe I use the original low-resolution images as network input.

I wonder what's the upscaling factor in your task.
And the low-resolution image you showed here seems a bicubic-interpolated form of the original LR image. I don't know whether this was caused by the image compression on the website or other reasons.

**Comment by qc17-THU** [comment]
> Thank you so much for taking the time to respond to me.

You're welcome :)

**Comment by mahirahzaini** [comment]
The upscaling factor is 2. As for my low-resolution image, I have shown below how I acquired the image.
![Picture1](https://user-images.githubusercontent.com/84991393/193128158-13fdb301-5892-4361-8300-1d98e6f13de5.png)

**Comment by qc17-THU** [comment]
Which interpolation methods did you use to generate your LR images? Nearest?

**Comment by mahirahzaini** [comment]
No interpolation was used to generate the LR images. I only use cropping method to generate the LR images and use the provided matlab code to generate the training dataset.

**Comment by mahirahzaini** [comment]
Hi there, I just wanted to double check in your paper CCPs dataset was trained up to 500000 iterations for DFCAN?

**Comment by qc17-THU** [comment]
Hi, actually 250,000~30,0000 iterations were enough. Then the model with the least validation loss was used for evaluation.
